mlx/ops.cpp

```diff
@@ -300,19 +300,27 @@ array reshape(
       size *= shape[i];
     }
   }
+
+  // Infer the shape
   if (size > 0) {
     auto q_and_r = std::ldiv(a.size(), size);
     if (infer_idx >= 0) {
       shape[infer_idx] = q_and_r.quot;
       size *= q_and_r.quot;
     }
+  } else if (infer_idx >= 0) {
+    std::ostringstream msg;
+    throw std::invalid_argument("Cannot infer the shape of an empty array");
```

Comment:
Nitty nit:

```suggestion
    throw std::invalid_argument("[reshape] Cannot infer the shape of an empty array");
```

Reply:
Yeah and also forgot the `ostringstream`. Let me fix all these...